Fix @hooks barrel import failing tsc module resolution

## Context

Surfaced during review of #36 (OIDC deploy migration PR). Confirmed pre-existing on `main` via a baseline `tsc --noEmit` check before that branch existed — unrelated to that PR's scope (workflow YAML only), so left untouched there and filed here instead, per the user's request to keep the repo fully clean (no TS errors, no failing tests, no lint issues).

## What

`pnpm exec tsc --noEmit` fails with:
```
src/pages/Pokedex/Pokedex.test.tsx(56,54): error TS2307: Cannot find module '@hooks' or its corresponding type declarations.
src/pages/Pokedex/Pokedex.tsx(5,102): error TS2307: Cannot find module '@hooks' or its corresponding type declarations.
```

**Root cause identified**: `vite.config.ts` defines a bare alias `'@hooks': '/src/hooks'`, which Vite/Vitest resolves against `src/hooks/index.ts` at runtime — this is why `pnpm build` and `pnpm test` both succeed despite the bare `import { ... } from '@hooks'` in `Pokedex.tsx:5` and the `vi.mock('@hooks', ...)` in `Pokedex.test.tsx:56`. But `tsconfig.json`'s `paths` only defines the wildcard `"@hooks/*": ["src/hooks/*"]` (requiring a subpath after `@hooks/`), with no bare `"@hooks"` entry — so `tsc`'s path resolution, unlike Vite's, can't resolve the barrel-style import with no subpath.

Every other `@`-aliased import in this codebase uses a subpath (e.g. `@components/ErrorBoundary`), so this is the only place the gap shows up.

## Why

Repo currently has zero build-breaking issues (lint clean, 88/88 tests pass) except this one `tsc` failure — closing it gets the repo to a fully clean state (no TS errors, no failing tests, no lint issues), which the user explicitly wants maintained going forward.

## Acceptance Criteria
- [x] `tsconfig.json`'s `paths` includes a bare `"@hooks": ["src/hooks/index.ts"]` entry alongside the existing `"@hooks/*": ["src/hooks/*"]` wildcard, matching `vite.config.ts`'s alias behavior
- [x] `pnpm exec tsc --noEmit` passes with zero errors
- [x] `pnpm test` and `pnpm lint` still pass (no behavior change — this is a type-resolution config fix only)

## Out of scope
Changing the import style in `Pokedex.tsx`/`Pokedex.test.tsx` to use a subpath instead (e.g. `@hooks/index`) — the barrel-import style is intentional and used elsewhere in spirit; the config should support it, not the other way around.

## Dependencies
None — standalone config fix, unrelated to any other open issue or milestone.

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "pokedex",
-  "version": "1.2.2",
+  "version": "1.2.3",
   "private": true,
   "type": "module",
   "scripts": {

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -13,6 +13,7 @@
     "paths": {
       "@api/*": ["src/api/*"],
       "@components/*": ["src/components/*"],
+      "@hooks": ["src/hooks/index.ts"],
       "@hooks/*": ["src/hooks/*"],
       "@pages/*": ["src/pages/*"],
       "@models/*": ["src/types/*"],

